drivers/crypto/crypto_si32.c

```diff
@@ -0,0 +1,1157 @@
+{
+	switch (ctx->keylen) {
+	case 32:
+		SI32_AES_A_select_key_size_256(SI32_AES_0);
+		break;
+	case 24:
+		SI32_AES_A_select_key_size_192(SI32_AES_0);
+		break;
+	case 16:
+		SI32_AES_A_select_key_size_128(SI32_AES_0);
+		break;
+	default:
+		LOG_ERR("Invalid key len: %" PRIu16, ctx->keylen);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static void assert_dma_settings_common(struct SI32_DMADESC_A_Struct *channel_descriptor)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
+
+	__ASSERT(channel_descriptor->CONFIG.SRCSIZE == 2,
+		 "Source size (SRCSIZE) and destination size (DSTSIZE) are 2 for a word transfer.");
+	__ASSERT(channel_descriptor->CONFIG.DSTSIZE == 2,
+		 "Source size (SRCSIZE) and destination size (DSTSIZE) are 2 for a word transfer.");
+	__ASSERT(channel_descriptor->CONFIG.RPOWER == 2,
+		 "RPOWER = 2 (4 data transfers per transaction).");
+}
+
+static void assert_dma_settings_channel_rx(struct SI32_DMADESC_A_Struct *channel_descriptor,
+					   struct cipher_pkt *pkt)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
+
+	assert_dma_settings_common(channel_descriptor);
+
+	__ASSERT(channel_descriptor->SRCEND.U32 == (uintptr_t)&SI32_AES_0->DATAFIFO,
+		 "Source end pointer set to the DATAFIFO register.");
+	__ASSERT(channel_descriptor->CONFIG.DSTAIMD == 0b10,
+		 "The DSTAIMD field should be set to 010b for word increments.");
+	__ASSERT(channel_descriptor->CONFIG.SRCAIMD == 0b11,
+		 "The SRCAIMD field should be set to 011b for no increment.");
+}
+
+static void assert_dma_settings_channel_tx(struct SI32_DMADESC_A_Struct *channel_descriptor)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
+
+	assert_dma_settings_common(channel_descriptor);
+
+	__ASSERT(channel_descriptor->DSTEND.U32 == (uintptr_t)&SI32_AES_0->DATAFIFO,
+		 "Destination end pointer set to the DATAFIFO register.");
+	__ASSERT(channel_descriptor->CONFIG.DSTAIMD == 0b11,
+		 "The DSTAIMD field should be set to 011b for no increment.");
+	__ASSERT(channel_descriptor->CONFIG.SRCAIMD == 0b10,
+		 "The SRCAIMD field should be set to 010b for word increments.");
+}
+
+static void assert_dma_settings_channel_xor(struct SI32_DMADESC_A_Struct *channel_descriptor)
+{
+	if (!IS_ENABLED(CONFIG_ASSERT)) {
+		ARG_UNUSED(channel_descriptor);
+	}
+
+	assert_dma_settings_common(channel_descriptor);
+
+	__ASSERT(channel_descriptor->DSTEND.U32 == (uintptr_t)&SI32_AES_0->XORFIFO,
+		 "Destination end pointer set to the XORFIFO register.");
+	__ASSERT(channel_descriptor->CONFIG.DSTAIMD == 0b11,
+		 "The DSTAIMD field should be set to 011b for no increment.");
+	__ASSERT(channel_descriptor->CONFIG.SRCAIMD == 0b10,
+		 "The SRCAIMD field should be set to 010b for word increments.");
+}
+
+/* Set up and start input (TX) DMA channel */
+static int crypto_si32_dma_setup_tx(struct cipher_pkt *pkt, uint_fast8_t in_buf_offset)
+{
+	struct dma_block_config dma_block_cfg = {0};
```

Comment:
You can just write `{ }` rather than `{0}`.

Reply:
Done